File: traditional_TP_methods/MainTestTwoPhaseAlgorithm_saveToFile.java
package traditional_TP;

import java.io.IOException;
import java.io.UnsupportedEncodingException;
import java.net.URL;

import traditional_TP.AlgoTwoPhase;
import traditional_TP.ItemsetsTP;
import traditional_TP.UtilityTransactionDatabaseTP;

/**
 * Example of how to use the TWOPhase Algorithm in source code.
 * @author Philippe Fournier-Viger, 2010
 */
public class MainTestTwoPhaseAlgorithm_saveToFile {

	public static void main(String [] arg) throws IOException{
		
		String input = fileToPath("new_DB_Utility.txt");
		String output = "D:/Eclipse/highUtility/src/two_phase/output.txt";

		double utilityFactor = 1; 
		
		// Loading the database into memory
		UtilityTransactionDatabaseTP database = new UtilityTransactionDatabaseTP();
		database.loadFile(input);
		
		//int min_utility = (int) (database.size() * utilityFactor);
		int min_utility = 20;
				
		// Applying the Two-Phase algorithm
		AlgoTwoPhase twoPhase = new AlgoTwoPhase();
		ItemsetsTP highUtilityItemsets = twoPhase.runAlgorithm(database, min_utility);
		
		highUtilityItemsets.saveResultsToFile(output, database.getTransactions().size());

		twoPhase.printStats();

	}

	public static String fileToPath(String filename) throws UnsupportedEncodingException{
		URL url = MainTestTwoPhaseAlgorithm_saveToFile.class.getResource(filename);
		 return java.net.URLDecoder.decode(url.getPath(),"UTF-8");
	}
}

File: traditional_TP_methods/main_test_two_phase_algorithm_save_to_file.py
"""Standalone driver for executing the Two-Phase algorithm and saving results."""

from __future__ import annotations

from pathlib import Path

from algo_two_phase import AlgoTwoPhase
from utility_transaction_database_tp import UtilityTransactionDatabaseTP


def file_to_path(filename: str) -> Path:
    return Path(__file__).resolve().parent / filename


def main() -> None:
    input_path = file_to_path("new_DB_Utility.txt")
    output_path = Path("two_phase_output.txt")

    database = UtilityTransactionDatabaseTP()
    database.load_file(input_path)

    min_utility = 20
    algo = AlgoTwoPhase()
    high_utility_itemsets = algo.run_algorithm(database, min_utility)

    high_utility_itemsets.save_results_to_file(
        output_path, len(database.get_transactions())
    )
    algo.print_stats()


if __name__ == "__main__":
    main()
